db/c.cc

```diff
@@ -506,6 +506,27 @@ const char* rocksdb_table_properties_get_user_collected_property(
   return nullptr;
 }
 
+const char** rocksdb_table_properties_get_user_collected_property_keys(
+    const rocksdb_table_properties_t* table_properties, size_t* key_count) {
+  auto properties = table_properties->rep->user_collected_properties;
```

Comment:
What's the type of properties? can you spell it out instead of auto?

Reply:
What I believe is happening is that you are copy-constructing the map in `properties` and that's why properties is immediately freed when returning from this function.

Reply:
What's the general rule here? Should I have used an explicit type for the iteration below for example? Is it simply because it's non-obvious what type it is here, that it's preferable to spell it out?

Reply:
Yes, best to spell it out, but what I'm wondering is whether you need to copy the map rather than iterating over existing one.

Reply:
Updated! The copied type was clear once I removed the `auto`, thanks for that!